src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -706,7 +708,52 @@ ActiveDataset::ActiveDataset(ThreadNetif &aThreadNetif):
 
 void ActiveDataset::StartLeader(void)
 {
+    if (mLocal.GetTimestamp() == NULL)
+    {
+        otOperationalDataset dataset;
+
+        memset(&dataset, 0, sizeof(dataset));
+
+        // Active Timestamp
+        dataset.mActiveTimestamp = 0;
+        dataset.mIsActiveTimestampSet = true;
+
+        // Channel
+        dataset.mChannel = mNetif.GetMac().GetChannel();
+        dataset.mIsChannelSet = true;
+
+        // Extended PAN ID
+        memcpy(dataset.mExtendedPanId.m8, mNetif.GetMac().GetExtendedPanId(), sizeof(dataset.mExtendedPanId));
+        dataset.mIsExtendedPanIdSet = true;
+
+        // Mesh-Local Prefix
+        memcpy(dataset.mMeshLocalPrefix.m8, mNetif.GetMle().GetMeshLocalPrefix(), sizeof(dataset.mMeshLocalPrefix));
+        dataset.mIsMeshLocalPrefixSet = true;
+
+        // Master Key
+        const uint8_t *key;
+        uint8_t keyLength;
+
+        key = mNetif.GetKeyManager().GetMasterKey(&keyLength);
+        memcpy(dataset.mMasterKey.m8, key, keyLength);
+        dataset.mIsMasterKeySet = true;
+
+        // Network Name
+        const char *name;
+
+        name = mNetif.GetMac().GetNetworkName();
+        memcpy(dataset.mNetworkName.m8, name, strlen(name));
+        dataset.mIsNetworkNameSet = true;
+
+        // Pan ID
+        dataset.mPanId = mNetif.GetMac().GetPanId();
+        dataset.mIsPanIdSet = true;
+
+        mLocal.Set(dataset);
+    }
+
     mNetwork = mLocal;
+    mNetwork.Store();
```

Comment:
`mNetwork` and `mLocal` are the same here, so I think this ok.  But in general, I think we should be storing the `mLocal` copy.